src/hotspot/share/utilities/vmError.cpp

```diff
@@ -1629,27 +1628,42 @@ void VMError::report_and_die(int id, const char* message, const char* detail_fmt
         // Watcherthread is about to call os::die. Lets just wait.
         os::infinite_sleep();
       } else {
-        // Crash or assert during error reporting. Lets continue reporting with the next step.
-        stringStream ss(buffer, sizeof(buffer));
+        // A secondary error happened. Print brief information, but take care, since crashing
+        // here would just recurse endlessly.
+        // Any information (signal, context, siginfo etc) printed here should use the function
+        // arguments, not the information stored in *this, since those describe the primary crash.
+        char tmp[256]; // cannot use global scratch buffer
```

Comment:
Is there any problems with making this static? Given that we care about the stack depth when we have repeated crashes.

Reply:
I think not. At this point we are single threaded. I'll convert to static.